jcl/src/java.base/share/classes/java/lang/J9VMInternals.java

```diff
@@ -503,6 +503,49 @@ static int fastIdentityHashCode(Object anObject) {
 	 */
 	static native int identityHashCode(Object anObject);
 
+	/*[IF INLINE-TYPES] */
+	/**
+	 * Answers an integer hash code for the parameter.
+	 * The caller must ensure that the parameter is a value type.
+	 * The hash code returned is the same one that would
+	 * be returned by java.lang.Object.hashCode(), assuming
+	 * the object's class has not overridden hashCode() and
+	 * that the parameter is a value type.
+	 *
+	 * @param		anObject	the object
+	 * @return		the hash code for the object
+	 * @throws		InternalError if the object is not a value type
+	 *
+	 * @see			java.lang.Object#hashCode
+	 */
+	static int valueHashCode(Object anObject) {
+		int hashcode;
+
+		try {
+			Class<?> valueObjectMethods = Class.forName("java.lang.runtime.ValueObjectMethods");
+			Method valueObjectHashCode = valueObjectMethods.getDeclaredMethod("valueObjectHashCode", Object.class);
+			valueObjectHashCode.setAccessible(true);
+			hashcode = (Integer)valueObjectHashCode.invoke(null, anObject);
+		} catch (ClassNotFoundException | NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+			throw new InternalError(e);
+		}
+
+		if (positiveOnlyHashcodes()) {
+			hashcode = Math.abs(hashcode);
```

Comment:
This does not match the behavior of the [native code](https://github.com/eclipse-openj9/openj9/blob/master/runtime/oti/ObjectHash.hpp#L186), it should be:
```
		hashcode &= 0x7FFF_FFFF;
```

Reply:
updated in 97c372b